### ISSUE TITLE
Slice out of range error when try to get types/struct

### ISSUE THREAD

**Issue by Li4n0** [goretk/gore#13]
I test redress today and get this error:
![image](https://user-images.githubusercontent.com/34324462/71225985-29c0ad80-2315-11ea-93a1-781e5a5fa921.png)

redress can get compiler version and src correctly,  only problems when trying to get the struct/types
![image](https://user-images.githubusercontent.com/34324462/71226090-9c318d80-2315-11ea-9b34-b4392239210e.png)

The binary I used for test is  [bettercap](https://github.com/bettercap/bettercap)

**Comment by TcM1911** [comment]
Reproduced:
```
panic: runtime error: slice bounds out of range [18446744073692771816:6638624]

goroutine 1 [running]:
github.com/goretk/gore.typeParse(0xc0009b8060, 0xc0000f2a20, 0xfffffffffefff5e8, 0xc001a4c020, 0x654c20, 0x654c20, 0x1000a20, 0x0)
	github.com/goretk/gore@v0.8.3/type.go:321 +0x38d1
github.com/goretk/gore.typeParse(0xc0009b8060, 0xc0000f2a20, 0x129c60, 0xc001a4c020, 0x654c20, 0x654c20, 0x1000a20, 0x0)
	github.com/goretk/gore@v0.8.3/type.go:618 +0x1215
github.com/goretk/gore.typeParse(0xc0009b8060, 0xc0000f2a20, 0xeed00, 0xc001a4c020, 0x654c20, 0x654c20, 0x1000a20, 0x0)
	github.com/goretk/gore@v0.8.3/type.go:402 +0x241b
github.com/goretk/gore.getTypes(0xc0000f2a20, 0x5becc0, 0xc00009a030, 0x0, 0x522580, 0xc0000d6140)
	github.com/goretk/gore@v0.8.3/type.go:76 +0x287
github.com/goretk/gore.(*GoFile).GetTypes(0xc0000d6140, 0x0, 0x0, 0x0, 0x0, 0x0)
	github.com/goretk/gore@v0.8.3/file.go:262 +0x52
main.standalone()
	github.com/goretk/redress@/standalone.go:59 +0x2d6
main.main()
	github.com/goretk/redress@/main.go:71 +0x4c
```

**Comment by TcM1911** [comment]
@Li4n0, I have identified what the issue is. The current parsing logic does not handle functions that have methods correctly. To fix this I would like to make some major refactoring of the logic first. This will take some time.

I have a workaround to fix the out of bounds read and gracefully handle function types with methods. It will not be able to identify the function argument types and return types but it will not panic.

Is the workaround solution acceptable for now?

**Comment by Li4n0** [comment]
@TcM1911  Yes it's acceptable. Thanks for your reply.

**Comment by TcM1911** [comment]
This was fixed with #15. A new release 0.8.4 was released with this fix (https://github.com/goretk/gore/releases/tag/v0.8.4).

Redress was also updated to 0.6.3 that pulls in this fix.
https://github.com/goretk/redress/releases/tag/v0.6.3
